src/vos/vos_io.c

```diff
@@ -717,12 +717,23 @@ vos_ioc_create(daos_handle_t coh, daos_unit_oid_t oid, bool read_only,
 		if (ioc->ic_size_fetch)
 			continue;
 
+		tot_iov_nr += iov_nr;
+		/* Generate an error message when there are too many IOVs for single IOD */
+		if (iov_nr >= 1024)
```

Comment:
Is this something we should prevent in our client by breaking things up?

Reply:
Yes, I think it would be nice if client is able to do that, but client can't handle all cases, for example, a simple fetch could have to read many non-contiguous physical extents on server.